Add FreeBSD start page (#64)
openzfs/openzfs-docs: docs/Getting Started/FreeBSD.rst

FreeBSD
=======

|ZoF-logo|

Installation on FreeBSD
-----------------------

> OpenZFS is available in the FreeBSD ports tree as sysutils/openzfs and

Reviewer: We should clarify that OpenZFS is also in the FreeBSD base system now on FreeBSD 13.0-CURRENT and later.

Author: Done

sysutils/openzfs-kmod. It can be installed on FreeBSD stable/12 or
later.

The ZFS utilities will be installed in /usr/local/sbin/, so make sure
your PATH gets adjusted accordingly.

To load the module at boot, put ``openzfs_load="YES"`` in
/boot/loader.conf, and remove ``zfs_load="YES"`` if migrating a ZFS
install.

Beware that the FreeBSD boot loader does not allow booting from root
pools with encryption active (even if it is not in use), so do not try
encryption on a pool you boot from.

Development on FreeBSD
----------------------

The following dependencies are required to build OpenZFS on FreeBSD:

-  FreeBSD sources in /usr/src or elsewhere specified by SYSDIR in env.
   If you don't have the sources installed you can install them with
   svnlite.

   Install source For FreeBSD 12:
   ::

     svnlite checkout https://svn.freebsd.org/base/stable/12 /usr/src

   Install source for FreeBSD Current:
   ::

     svnlite checkout https://svn.freebsd.org/base/head/ /usr/src

-  Packages for build:
   ::

      pkg install \
          autoconf \
          automake \
          autotools \

>           bash \

Reviewer: We no longer need bash for building ZFS.

Author: Moved bash to tests requirements then, IIRC there were some on bash.

FreeBSD
=======

|ZoF-logo|

Installation on FreeBSD
-----------------------

> OpenZFS is available as:

Reviewer: ```suggestion
OpenZFS is available pre-packaged as:
```

> - part of FreeBSD base system from FreeBSD 13.0-CURRENT

Reviewer: ```suggestion
- the zfs-2.0-release branch, in the FreeBSD base system from FreeBSD 13.0-CURRENT forward
```

> - ports as sysutils/openzfs and sysutils/openzfs-kmod prior to FreeBSD 12.

Reviewer: ```suggestion
- the master branch, in the FreeBSD ports tree as sysutils/openzfs and sysutils/openzfs-kmod from FreeBSD 12.1 forward
```

> 

Reviewer: ```suggestion

The rest of this document describes the use of OpenZFS either from ports/pkg or built manually from sources for development.

```

The ZFS utilities will be installed in /usr/local/sbin/, so make sure
your PATH gets adjusted accordingly.

To load the module at boot, put ``openzfs_load="YES"`` in
/boot/loader.conf, and remove ``zfs_load="YES"`` if migrating a ZFS
install.

Beware that the FreeBSD boot loader does not allow booting from root
pools with encryption active (even if it is not in use), so do not try
encryption on a pool you boot from.

Development on FreeBSD
----------------------

The following dependencies are required to build OpenZFS on FreeBSD:

-  FreeBSD sources in /usr/src or elsewhere specified by SYSDIR in env.
   If you don't have the sources installed you can install them with
   svnlite.

   Install source For FreeBSD 12:
   ::

     svnlite checkout https://svn.freebsd.org/base/stable/12 /usr/src

   Install source for FreeBSD Current:
   ::

>      svnlite checkout https://svn.freebsd.org/base/head/ /usr/src

Reviewer: ```suggestion
     svnlite checkout https://svn.freebsd.org/base/head /usr/src
```

-  Packages for build:
   ::

      pkg install \
          autoconf \
          automake \
          autotools \
          git \
          gmake \

-  Optional packages for build:
   ::

      pkg install python37 # or your preferred Python version

>       pkg install py37-sysctl # needed for arcstat

Reviewer: ```suggestion
      pkg install py37-sysctl # needed for arcstat, arc_summary, dbufstat
```

-  Packages for checks and tests:
   ::

      pkg install \
          base64 \
          bash \
          checkbashisms \
          fio \
          hs-ShellCheck \
          ksh93 \
          pamtester \
          py37-flake8 \
          python37 \
          sudo

   Your preferred python version may be substituted. The user for
   running tests must have NOPASSWD sudo permission.

To build and install:

::

   # as user
   git clone https://github.com/openzfs/zfs
   cd zfs
   ./autogen.sh
   ./configure
   gmake -j`sysctl -n hw.ncpu`
   # as root
   gmake install

> To use OpenZFS kernel modules when FreeBSD starts edit file ``/boot/loader.conf`` :

Reviewer: ```suggestion
To use the OpenZFS kernel module when FreeBSD starts, edit ``/boot/loader.conf`` :
```

> Replace line:

Reviewer: ```suggestion
Replace the line:
```

::

  zfs_load="YES"

with:

::

>   openzfs_load="Yes"

Reviewer: ```suggestion
  openzfs_load="YES"
```

> The stock FreeBSD ZFS binaries are installed in /sbin.  OpenZFS binaries are installed to /usr/local/sbin.  To use OpenZFS binaries adjust your path so /usr/local/sbin is listed before /sbin.  Otherwise the native ZFS binaries will be used.

Reviewer: ```suggestion
The stock FreeBSD ZFS binaries are installed in /sbin.  OpenZFS binaries are installed to /usr/local/sbin when installed form ports/pkg or manually from the source.  To use OpenZFS binaries, adjust your path so /usr/local/sbin is listed before /sbin.  Otherwise the native ZFS binaries will be used.
```
